### PR TITLE
1761 create a new color dropdown component

### LINKED ISSUE #1761: [ColorDropdown] Create a new ColorDropdown component
Part of issue #1753

### What package is this for?

- [x] React
- [ ] Angular

### Summary
A color dropdown component is need for the HotSpotEditorModal. 

**Additional context** 
image

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #1761

**Summary**
A color dropdown component for the HotSpotEditorModal.

**Change List (commits, features, bugs, etc)**
- Extended the carbon Dropdown to show color squares per item and for the selection
- Added default colors
- Added option to pass in custom colors
- Added callback for when color selection change
- Added a new story
- Added unit tests
- Verified RL

**Acceptance Test (how to verify the PR)**
Check out the stories at
https://deploy-preview-1765--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-experimental-colordropdown--default-example

### CODE REVIEW

File: src/components/ColorDropdown/ColorDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  /** Internationalisation strings */
+  label: PropTypes.string,
+  titleText: PropTypes.string,
+  id: PropTypes.string.isRequired,
+  light: PropTypes.bool,
+  /** Callback for when any of the Dropdown color value changes */
+  onChange: PropTypes.func.isRequired,
+  selectedColor: colorPropType,
+  testID: PropTypes.string,
+};
```
Comment (tay1orjones) [inline src/components/ColorDropdown/ColorDropdown.jsx:39]: These comments are parsed by storybook and populated into the prop tables. Each one is individual, so we should be specific.

```suggestion
const propTypes = {
  colors: PropTypes.arrayOf(colorPropType),
  /** The label applied to the dropdown for internationalisation */
  label: PropTypes.string,
  /** The titleText applied to the dropdown for internationalisation */
  titleText: PropTypes.string,
  id: PropTypes.string.isRequired,
  light: PropTypes.bool,
  /** Callback for when any of the Dropdown color value changes */
  onChange: PropTypes.func.isRequired,
  selectedColor: colorPropType,
  testID: PropTypes.string,
};
```

**Review by tay1orjones — APPROVED** [review]
One minor item, otherwise looks great! 👏

**Comment by marcJV** [comment]
image
Just a visual bug here. The bottom border should have a gap at each side of the item. I know this was a fix so that when using itemToString we don't show `[object Object]` while hovering.

File: src/components/ColorDropdown/ColorDropdown.jsx (hunk 1 above)
Comment (bjornalm) [reply src/components/ColorDropdown/ColorDropdown.jsx:39]: Yes, of course. I was a bit too fast when externalising this one, I'll add proper comments to all properties.

**Comment by bjornalm** [comment]
> image
> Just a visual bug here. The bottom border should have a gap at each side of the item. I know this was a fix so that when using itemToString we don't show `[object Object]` while hovering.

Good catch. I've fixed this.

**Review by marcJV — APPROVED** [review]

**Pull request merged**